independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanInfo.java

```diff
@@ -449,9 +449,11 @@ private void putLifecycleInterceptors(Map<InterceptionType, InterceptionInfo> li
 
     private void addClassLevelBindings(ClassInfo classInfo, Collection<AnnotationInstance> bindings) {
         beanDeployment.getAnnotations(classInfo).stream()
-                .filter(a -> beanDeployment.getInterceptorBinding(a.name()) != null
-                        && bindings.stream().noneMatch(e -> e.name().equals(a.name())))
-                .forEach(a -> bindings.add(a));
+                .flatMap(a -> beanDeployment.extractInterceptorBindings(a).stream())
+                // TODO this is weird -- either we allow multiple class-level interceptor bindings of the same type,
```

Comment:
Isn't having multiple bindings of the same type (and all on class level) an error according to specification?

Reply:
That's how I read the spec, indeed -- but it doesn't look like we enforce it. (At least not yet.)

Reply:
As per Zulip conversation - this should be validated and the spec requires this because you can, for instance, declare the same binding with member on a class and on a stereotype that this class uses leading to a conflict.

Reply:
+1 for validation and potentially a build failure, i.e. repeatable bindings are only allowed on methods which should be documented here https://quarkus.io/guides/cdi-reference#nonstandard_features

Reply:
OK, only supporting repeatable interceptor bindings on methods, that's an interesting proposition. @manovotn does perhaps Weld do the same? That would explain why there's no test for repeatable interceptor bindings on classes.

Reply:
> does perhaps Weld do the same? That would explain why there's no test for repeatable interceptor bindings on classes.

We should find out ;-)

Reply:
Did you just volunteer to try that you? :-P

Reply:
A quick remake of the existing Weld test shows me that interception happens even if I put repeatable annotation on a class-level of the bean.

Reply:
Good to know. Doesn't necessarily mean we have to do the same, but I think the current behavior is actually pretty reasonable. (Once you make peace with the fact that whenever you put an interceptor binding on a method, all "inherited" interceptor bindings of the same type are ignored :-) )

Reply:
> Doesn't necessarily mean we have to do the same

No, but for most people familiar with CDI, it would then behave in an "expected manner". E.g. if they used it before, they likely did so on Weld and aligning it would provide consistency. So I'd be +1 for that.